testnet/src/node.rs

```diff
@@ -243,18 +374,41 @@ impl Node {
             _ => panic!("Unable to retrieve local peer")
         };
 
-        let p2p_net = PeerNetwork::new(peerdb, local_peer, TESTNET_PEER_VERSION, burnchain, view, self.config.connection_options.clone());
-        let rpc_sock = self.config.node.rpc_bind.parse()
-            .expect(&format!("Failed to parse socket: {}", &self.config.node.rpc_bind));
-        let p2p_sock = self.config.node.p2p_bind.parse()
-            .expect(&format!("Failed to parse socket: {}", &self.config.node.p2p_bind));
-        let _join_handle = spawn_peer(
+        let blocks_paths = self.chain_state.blocks_path.clone();
+
+        // now we're ready to instantiate a p2p network object, the relayer, and the event dispatcher
+
+        let mut p2p_net = PeerNetwork::new(peerdb, local_peer.clone(), TESTNET_PEER_VERSION, burnchain, view,
+                                           self.config.connection_options.clone());
+
+        // setup the relayer channel
+        let (relay_send, relay_recv) = channel();
+        self.relay_channel = Some(relay_send.clone());
+
+        if spawn_relayer_miner {
+            let relayer = Relayer::from_p2p(&mut p2p_net);
+
+            spawn_miner_relayer(relayer, local_peer,
+                                self.config.clone(), &self.keychain,
+                                self.config.get_burn_db_file_path(),
+                                self.config.get_chainstate_path(),
+                                relay_recv, self.dispatcher_channel.clone());
+        } else {
+            // just no-op on the relayer events
+            thread::spawn(move || {
+                while let Ok(_) = relay_recv.recv() {
+                }
+            });
+        }
+
+        spawn_peer(
             p2p_net, 
             &p2p_sock, 
             &rpc_sock, 
             self.config.get_burn_db_file_path(),
             self.config.get_chainstate_path(), 
-            5000).unwrap();
+            5000, relay_send).unwrap();
```

Comment:
Should the 5000 be a setting? Constant?

Reply:
Probably a constant -- poll timeouts sometimes are and sometimes aren't configurable options. If reducing the poll timeout improves performance, there's usually something wrong in the implementation. The reason you might want to reduce it is that it will eventually impact how quickly the thread might respond to a SIGTERM.

Reply:
Pretty sure this is an interruptable poll under the hood, so it should respond to SIGINT and SIGTERM (but please let me know if not).